### PR TITLE
Implementation of INET6_NTOA

### PR DESCRIPTION
**Pull request by guilhermehubner**
closes https://github.com/pingcap/tidb/issues/3305

### CODE REVIEW

File: expression/builtin_miscellaneous_test.go [hunk 1]
```diff
@@ -193,6 +193,41 @@ func (s *testEvaluatorSuite) TestInetNtoa(c *C) {
...
+			0x02, 0x03, 0x04}, "::ffff:1.2.3.4"},
+		{[]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF,
+			0xFF, 0xFF, 0xFF}, "::ffff:255.255.255.255"},
+		{[]byte{}, nil},
+		{[]byte{0x0A, 0x00, 0x05}, nil},
+		{[]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0xFF,
+			0xFF, 0xFF}, nil},
+	}
+	fc := funcs[ast.Inet6Ntoa]
+	for _, test := range tests {
```
Comment (zhexuany) [inline expression/builtin_miscellaneous_test.go]: Please add a mean-to-be-failed test. We need catch any malformed ip address.

Comment (guilhermehubner) [inline expression/builtin_miscellaneous_test.go]: I don't understood your request.

I've added some fail cases like these:

```go
{[]byte{0x0A, 0x00, 0x05}, nil} // missing a byte ipv4
{[]byte{}, nil} // missing bytes
{[]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF, 
0xFF, 0xFF, 0xFF, 0xFF}, nil} // missing a byte ipv6
```

What cases you think is missing?

**Comment by tiancaiamao** [comment]
LGTM
Good job! @guilhermehubner

File: expression/builtin_miscellaneous_test.go (hunk 1 above)
Comment (zhexuany) [inline expression/builtin_miscellaneous_test.go]: Ah, my bad. I did not recognize these failed cases. Is it better if you add these comments? BTW, awesome work. 👍

Comment (guilhermehubner) [inline expression/builtin_miscellaneous_test.go]: @zhexuany agreed. Comments added :grin:

**Comment by shenli** [comment]
@guilhermehubner Please update [typeinferer.go](https://github.com/pingcap/tidb/blob/master/expression/typeinferer.go#L298) to add type inference result and also add unit test in [typeinferer_test.go](https://github.com/pingcap/tidb/blob/master/expression/typeinferer_test.go#L328).

**Comment by guilhermehubner** [comment]
@shenli I'm not sure about my last changes (typeinferer.go & typeinferer_test.go). Could you help take a look and see if it has any mistakes and help me to solve them?

**Comment by shenli** [comment]
LGTM
@guilhermehubner Well done!

**Pull request merged**